scripts/cloudbuild.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python
+'''creates cloudbuild.json for caliban'''
+
+from absl import flags, app, logging
+from typing import Dict, List, Any, NamedTuple, Optional
+
+import json
+import pprint as pp
+
+# ----------------------------------------------------------------------------
+_CLOUD_BUILDER = 'gcr.io/cloud-builders/docker'
+_BASE_URL = 'gcr.io/blueshift-playground/blueshift'
+_GPU_BASE_TAG = 'base'
+_CPU_TAG = 'cpu'
+_GPU_TAG = 'gpu'
+_CPU_BASE_IMAGE = 'ubuntu:18.04'
+_CPU_DOCKERFILE = 'dockerfiles/Dockerfile'
+_GPU_DOCKERFILE = 'dockerfiles/Dockerfile.gpu'
+
+# ----------------------------------------------------------------------------
+FLAGS = flags.FLAGS
+
+flags.DEFINE_string('output', None, 'output json file')
+flags.DEFINE_string('config', None, 'input configuration file')
+flags.DEFINE_string('base_url', _BASE_URL, 'base url for images')
+flags.DEFINE_integer('timeout_sec', 24 * 60 * 60,
+                     'timeout for all builds to complete')
+
+# ----------------------------------------------------------------------------
+logging.get_absl_handler().setFormatter(None)
+
+Config = Dict[str, Any]
+
+
+# ----------------------------------------------------------------------------
+class BuildStep(NamedTuple):
+  '''a google cloud-build step'''
+  sid: str
+  name: str = _CLOUD_BUILDER
+  args: List[str] = []
+  dependencies: List[str] = ['-']
+
+  def to_dict(self) -> Dict[str, Any]:
+    '''converts step to dict'''
+    return {
+        'id': self.sid,
+        'name': self.name,
+        'args': self.args,
+        'waitFor': self.dependencies,
+    }
+
+
+# ----------------------------------------------------------------------------
+class ImageSpec(NamedTuple):
+  '''a caliban base image spec'''
+  dockerfile: str = _CPU_DOCKERFILE
+  base_image: Optional[str] = _CPU_BASE_IMAGE
+  cpu_version: Optional[str] = None
+  gpu_version: Optional[str] = None
+  build_args: List[str] = []
+
+  def tag(self) -> str:
+    '''returns tag for this spec'''
+    parts = []
+    if self.gpu_version is None:  # cpu image
```

Comment:
Can we pull this logic out into a separate function, and return some enum value? It is always easier for me to read these conditionals when they're simply defining a mode. Then we can switch-case on the mode inside here (and anywhere else that needs it).